src/coreclr/jit/llvmcodegen.cpp

```diff
@@ -1835,6 +1907,20 @@ void Llvm::buildBoundsCheck(GenTreeBoundsChk* boundsCheckNode)
     emitJumpToThrowHelper(indexOutOfRangeValue, boundsCheckNode->gtThrowKind);
 }
 
+void Llvm::buildCkFinite(GenTreeUnOp* ckNode)
+{
+    assert(varTypeIsFloating(ckNode));
+    Type* fpLlvmType = getLlvmTypeForVarType(ckNode->TypeGet());
+    Value* opValue = consumeValue(ckNode->gtGetOp1(), fpLlvmType);
+
+    // Taken from IR Clang generates for "isfinite".
+    Value* absOpValue = _builder.CreateIntrinsic(llvm::Intrinsic::fabs, fpLlvmType, opValue);
+    Value* isNotFiniteValue = _builder.CreateFCmpUEQ(absOpValue, llvm::ConstantFP::get(fpLlvmType, INFINITY));
```

Comment:
Is this doing the right thing for NaN?

Reply:
I see that there is a test case for this, so it presumably works.

Reply:
> Is this doing the right thing for NaN?

Yes, `fabs` will preserve the NaN and unordered compares return `true` if any one of the operands are `NaN`.

Reply:
I think we have tests for it, but looking at them I think this should be `||` 
https://github.com/SingleAccretion/runtimelab/blob/c59764fb6cef967f06edcf3deab73574f2b498f8/src/tests/nativeaot/SmokeTests/HelloWasm/HelloWasm.cs#L2714
I hope we are getting away with that as they are all false, so `||` or `&&` is coming to the same thing.